app/code/Magento/Integration/Model/CustomerTokenService.php

```diff
@@ -99,7 +109,11 @@ public function createCustomerAccessToken($username, $password)
                 )
             );
         }
-        $this->eventManager->dispatch('customer_login', ['customer' => $customerDataObject]);
+
+        $customerModel = $this->customerRegistry->retrieve($customerDataObject->getId());
+        $this->eventManager->dispatch('customer_login', ['customer' => $customerModel]);
```

Comment:
Changing the type of the object bound to the `customer_login` event would potentially break all the third-party implementations based on that event. Since this could be a breaking change, I'm not sure this can be released on 2.4, I will check.